skactiveml/base.py

```diff
@@ -179,6 +182,514 @@ def query(self, X_cand, *args, A_cand=None, batch_size=1,
+            The utilities provided by the stream-based active learning
+            strategy.
+        return_budget_left : bool,
+            If true, also return whether there was budget left for each
+            assessed utility.
+        simulate : bool,
+            If True, the internal state of the budget manager before and after
+            the query is the same.
+
+        Returns
+        -------
+        utilities: ndarray of shape (n_samples,)
+            Checked utilities
+        return_budget_left : bool,
+            Checked boolean value of `return_budget_left`.
+        simulate : bool,
+            Checked boolean value of `simulate`.
+        """
+        # Check if utilities is set
+        if not isinstance(utilities, np.ndarray):
+            raise TypeError(
+                "{} is not a valid type for utilities".format(type(utilities))
+            )
+        # Check return_utilities.
+        check_scalar(return_budget_left, "return_budget_left", bool)
+        # Check return_utilities.
+        check_scalar(simulate, "simulate", bool)
+        # Check budget
+        self._validate_budget(get_default_stream_budget())
+        return utilities, return_budget_left, simulate
+
+
+def get_default_stream_budget():
+    """This function defines the default budget which should be used when no
+    budget is provided by the user.
+
+        Returns
+        -------
+        default_budget: float
+            The default budget used by the user.
+        """
+    return 0.1
+
+
+class SingleAnnotStreamBasedQueryStrategy(QueryStrategy):
+    """Base class for all stream-based active learning query strategies in
+       scikit-activeml.
+
+    Parameters
+    ----------
+    budget_manager : BudgetManager
+        The BudgetManager which models the budgeting constraint used in
+        the stream-based active learning setting.
+
+    random_state : int, RandomState instance, default=None
+        Controls the randomness of the estimator.
+    """
+
+    def __init__(self, budget_manager, random_state=None):
+        super().__init__(random_state=random_state)
+        self.budget_manager = budget_manager
+
+    @abstractmethod
+    def query(
+        self, X_cand, *args, return_utilities=False, simulate=False, **kwargs
+    ):
+        """Ask the query strategy which instances in X_cand to acquire.
+
+        The query startegy determines the most useful instances in X_cand,
+        which can be acquired within the budgeting constraint specified by the
+        budget_manager.
+        Please note that, when the decisions from this function
+        may differ from the final sampling, simulate=True can set, so that the
+        query strategy can be updated later with update(...) with the final
+        sampling. This is especially helpful, when developing wrapper query
+        strategies.
+
+        Parameters
+        ----------
+        X_cand : {array-like, sparse matrix} of shape (n_samples, n_features)
```

Comment:
formatting of multiple options:  evtl. array-like | sparse matrix

Reply:
seems to be correct (correct in other files):
https://scikit-learn.org/stable/modules/generated/sklearn.cluster.k_means.html#sklearn.cluster.k_means